cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto

```diff
@@ -154,45 +279,100 @@ message PackageRepository {
 }
 
-// GetPackageRepositories
+// -- Start other definitions  -- 
+
+// Context
 //
-// Response for GetPackageRepositories
-message GetPackageRepositoriesResponse {
-  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
-    example: '{"repositories": [{"name": "repo-name.example.com", "namespace": "", "url": "foo.registry.example.com/repo-name/main@sha256:cecd9b51b1f29a773a5228fe04faec121c9fbd2969de55b0c3804269a1d57aa5"}]}'
-  };
-  // Repositories
+// A Context specifies the context of the message
+message Context {
+  // TODO: add example for API docs
+  // option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+  //   example: '{}'
+  // };
+  
+  // Cluster
   //
-  // List of PackageRepository
-  repeated PackageRepository repositories = 1;
+  // A cluster name can be provided to target a specific cluster if multiple
+  // clusters are configured, otherwise all clusters will be assumed.
+  string cluster = 1;
+
+  // Namespace
+  //
+  // A namespace can be provided if the packaging system allows namespaced packages.
+  string namespace = 2;
 }
 
-message PackageReference {
-  // required package name should be unique within the scope of specified repository, if specified
-  string name = 1; 
+// PackageRepositoryReference
+//
+// A PackageRepository is identified by a fully qualified name and
+// optionally a specific namespace.
+message PackageRepositoryReference {
+  // TODO: add example for API docs
+  // option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+  //   example: '{}'
+  // };
 
-  // A PackageRepository is identified by a fully qualified name and
-  // optionally a specific namespace.
-  message PackageRepositoryReference {
-    string name = 1;
-    string namespace = 2;
-  }
-  // An optional package repository reference where this package is located. Not
-  // all plugins may support this back reference (eg. kapp-controller)
-  PackageRepositoryReference repository = 2;
+  // Package repository name
+  //
+  // The fully qualified name of the package repository.
+  string name = 1;
+  
+  // Package repository namespace
+  //
+  // An optional namespace for namespaced package repositories.
+  string namespace = 2;
 }
 
+// AvailablePackageReference
 //
-// Request for GetPackageMeta
-message GetPackageMetaRequest {
-  PackageReference package = 1;
+// An AvailablePackageReference has the minimum information required to uniquely
+// identify an available package. This is re-used on the summary and details of an
+// available package.
+message AvailablePackageReference {
+  // TODO: add example for API docs
+  // option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+  //   example: '{}'
+  // };
+
+  // Available package context
+  //
+  // The context (cluster/namespace) for the package.
+  Context context = 1;
+
+  // Available package name
+  //
+  // The fully qualified name of the package (ie. a unique name in context).
+  string name = 2;
```

Comment:
so this should now be
```
string identifier = 2;
```
See design doc for example comment (feel free to improve it :) )